metroae

```diff
@@ -1806,6 +1837,317 @@ function container_main {
     fi
 }
 
+#################################################################################
+#                                   Plugins                                     #
+#################################################################################
+
+function copy_plugin_file_into_mount_point_if_necessary {
```

Comment:
Please add `debug ${FUNCNAME[0]}` to each of these new functions. They are *so* helpful when trying to debug/trace execution.